ferry.go

```diff
@@ -134,6 +134,15 @@ func (f *Ferry) Initialize() (err error) {
 		return err
 	}
 
+	isReplica, err := f.checkDbIsAReplica(f.TargetDB)
```

Comment:
All this method does is check if the DB is READ_ONLY, right? Is this a reliable measure of whether or not the DB is a replica? Should we look at something like SLAVE STATUS?

Reply:
SLAVE STATUS is actually not a great way. For example, Shopify runs a master-master configuration which means masters from different location are slaving off each other (and hence SLAVE STATUS will show the slave thread on both instances)

Reply:
Not for now, but perhaps it is a good idea to defer this task to WaitUntilReplica struct such that people can implement arbitrary ways to check this.

Reply:
yeah that seems desirable for non-shopify users. I'll open an issue and we'll take care of it when there's interest